File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int count_letters(string s);
int count_words(string s);
int count_sentences(string s);
void get_grade_level(string text);

int main(void)
{
    // Get input text from user
    string text = get_string("Text: ");
    // Print Coleman-Lau index
    get_grade_level(text);
}

// Count the number of letters in a string
int count_letters(string s)
{
    int count = 0;
    for(int i = 0, n = strlen(s); i < n; i++)
    {
        if(isalpha(s[i]))
        {
            count++;
        }
    }
    return count;
}

// Count the number of words in a string
int count_words(string s)
{
    int count = 1;
    for(int i = 0, n = strlen(s); i <n; i++)
    {
        if(isspace(s[i]))
        {
            count++;
        }
    }
    return count;
}

// Count the number of sentences in a string
int count_sentences(string s)
{
    int count = 0;
    for(int i = 0, n = strlen(s); i <n; i++)
    {
        if(isalpha(s[i-1]))
        {
            if(s[i] == '.' || s[i] == '!' || s[i] == '?')
            {
            count++;
            }
        }
    }
    return count;
}

void get_grade_level(string text)
{
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Calculate grade level using Coleman-Lau formula
    float l = ((float) letters / words) * 100.0;
    float s = ((float) sentences / words) * 100.0;
    int grade = round(.0588 * l - 0.296 * s - 15.8);

    // Print result
    if(grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if(grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string
import re


def main():
    # Get input text from user
    text = get_string('Text: ')
    # Print grade level (Coleman-Lau index)
    getGradeLevel(text)


def getGradeLevel(str):
    # Count the letters, words and sentences found in the string. The empty string at the end of each regex split is removed with - 1
    letters = len(re.split('[a-zA-Z]', str)) - 1
    words = len(str.split())
    sentences = len(re.split('[.!?]', str)) - 1

    # Calculate grade level using Coleman-Lau formula
    l = (letters / words) * 100
    s = (sentences / words) * 100
    grade = round(.0588 * l - 0.296 * s - 15.8)

    # Print the result
    if grade < 1:
        print('Before Grade 1')
    elif grade > 16:
        print('Grade 16+')
    else:
        print('Grade ', grade)


main()